### PR TITLE
Add 'count' test in L0_warmup

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Review by tanmayv25 — APPROVED** [review]

File: qa/L0_warmup/test.sh [hunk 1]
```diff
@@ -96,12 +97,18 @@ for BACKEND in ${BACKENDS}; do
         echo "    }" >> config.pbtxt && \
         echo "}]" >> config.pbtxt )
 
-    # zero data
+    # zero data. For realistic sequence model, 'repeat_count' may not work
+    # well because the model will expect a valid sequence of requests which
+    # should be represented by a series of warmup samples. 'repeat_count'
+    # essentially "resend" one of the sample, which may invalidate the sequence.
```
Comment (dyastremsky) [inline qa/L0_warmup/test.sh:103]: nit: resends

File: qa/L0_warmup/test.sh [hunk 2]
```diff
@@ -143,7 +150,12 @@ for BACKEND in ${BACKENDS}; do
         echo -e "\n***\n*** Failed. Expected warmup for stateless model\n***"
         RET=1
     fi
-    grep "is running warmup sample 'sequence sample'" $SERVER_LOG
+    grep "is running warmup sample 'sequence sample' for iteration 1" $SERVER_LOG
+    if [ $? -ne 0 ]; then
+        echo -e "\n***\n*** Failed. Expected warmup for stateful model\n***"
+        RET=1
+    fi
+    grep "is running warmup sample 'sequence sample' for iteration 2" $SERVER_LOG
```
Comment (dyastremsky) [inline qa/L0_warmup/test.sh:158]: Perhaps add which iteration failed in the echoed statement for quicker debugging.

**Review by rmccorm4 — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

**Comment by rmccorm4** [comment]
@GuanLuo Can be a followup doc ticket if you want, but can we add `repeat_count` mention/example in https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#model-warmup ?

**Comment by GuanLuo** [comment]
I added pointer to L0_warmup test so we don't need to update the doc for new warmup feature, and hopefully the way we specify the setting in L0_warmup is straight-forward

**Review by rmccorm4 — DISMISSED** [review]

**Review by CoderHam — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**